Mathlib/LinearAlgebra/TensorProduct/RightExactness.lean

```diff
@@ -0,0 +1,358 @@
+/-
+Copyright (c) 2023 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambetr-Loir
+-/
+
+import Mathlib.Algebra.Exact
+import Mathlib.RingTheory.TensorProduct
+
+/-! # Right exactness properties of tensor product
+* `TensorProduct.rTensor_exact` says that one can tensor a short exact sequence on the right,
+  and `rTensor.equiv` gives a LinearEquiv.
+
+* `TensorProduct.lTensor_exact` says that one can tensor a short exact sequence on the left,
+  and `lTensor.equiv` gives a LinearEquiv.
+
+* For `N : Submodule R M`, `LinearMap.exact_subtype_mkQ N` says that
+  the inclusion of the submodule and the quotient map form an exact pair,
+  and `lTensor_mkQ` compute `ker (lTensor Q (N.mkQ))` and similarly for `rTensor_mkQ`
+
+* `TensorProduct.map_ker` computes the kernel of `TensorProduct.map f g'`
+  in the presence of two short exact sequences.
+
+The proofs are those of [bourbaki1989] (chap. 2, §3, n°6)
+
+## TODO
+
+* Treat the noncommutative case
+
+* Treat the case of modules over semirings
+  (For a possible definition of an exact sequence of commutative semigroups, see
+  [Grillet-1969b], Pierre-Antoine Grillet,
+  *The tensor product of commutative semigroups*,
+  Trans. Amer. Math. Soc. 138 (1969), 281-293, doi:10.1090/S0002-9947-1969-0237688-1 .)
+
+* Treat algebras (further PR)
+-/
+
+
+section Modules
+
+open TensorProduct LinearMap
+
+section Semiring
+
+variable {R : Type*} [CommSemiring R]
+variable {M N P P' Q: Type*}
+  [AddCommMonoid M] [AddCommMonoid N] [AddCommGroup P] [AddCommMonoid P'] [AddCommMonoid Q]
+  [Module R M] [Module R N] [Module R P] [Module R P'] [Module R Q]
+
+variable {f : M →ₗ[R] N} (g : N →ₗ[R] P)
```

Comment:
```suggestion
variable {R : Type*} [CommSemiring R] {M N P P' Q: Type*}
  [AddCommMonoid M] [AddCommMonoid N] [AddCommGroup P] [AddCommMonoid P'] [AddCommMonoid Q]
  [Module R M] [Module R N] [Module R P] [Module R P'] [Module R Q]
  {f : M →ₗ[R] N} (g : N →ₗ[R] P)
```
Looks nicer to me, but not so important. (If you change this you can do the same everywhere else).